### PR TITLE
Added SessionViewModel for better state management

### PR DESCRIPTION
**Pull request by MorganSchepp**

### CODE REVIEW

File: app/src/main/java/com/rrc/adev3007/pixel_perfect/the_y_app/HomeActivity.kt [hunk 1]
```diff
@@ -35,81 +36,86 @@ import com.rrc.adev3007.pixel_perfect.the_y_app.pages.Dislikes
...
-            Surface(
-                modifier = Modifier.fillMaxSize(),
-                color = if(session!!.getBoolean("darkMode", false)) Color.Black
-                    else Color.White
-                ) {
+    @Composable
+    fun HomeScreen(viewModel: SessionViewModel) {
+        val darkMode by viewModel.darkMode
+        val username by viewModel.username
+        var herbAderb by remember { mutableStateOf(1) }
```
Comment (DZ-FSDev) [inline app/src/main/java/com/rrc/adev3007/pixel_perfect/the_y_app/HomeActivity.kt:54]: This must've been an artefact from earlier testing...

**Review by DZ-FSDev — APPROVED** [review]
Nice. This refactor reduces cognitive complexity further through abstraction of the Session. Good job!

**Comment by mbialowas10** [comment]
Fellas, not sure what the root issue here was other than adding VM to save state across app.

**Comment by MorganSchepp** [comment]
yeah this was a refactor of our session management. no debug for this one,
just a refactoring.

**Comment by MorganSchepp** [comment]
i guess there was kind of an issue Michael, we couldn’t access the session
context without passing in our application context, which can only be used
at the activity level, not the composable level, so i made a VM to make
accessing the data a bit easier

**Pull request merged**